src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -65,6 +66,21 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.STR_DPKG_WAS_INTERRUPTED = "E: dpkg was interrupted, you must manually run 'sudo dpkg --configure -a' to correct the problem."
         self.ESM_MARKER = "The following packages could receive security updates with UA Infra: ESM service enabled:"
 
+        # pro client pre-requisite checks.
+        try:
+            self.maximum_os_version_supported = 18  # Currently we support only ubuntu 18 or lesser versions.
```

Comment:
Have an enum in Constants like this:

class UbuntuProClientSettings(EnumBackport):
     FEATURE_ENABLED = True
     MAX_OS_MAJOR_VERSION_SUPPORTED = 18
... anything else that qualifies ...